### ISSUE TITLE
Adapting for ResNet18

### ISSUE THREAD

**Issue by terrainer** [DrWuHonglin/CMTFNet#1]
Hi,

I'm trying to adapt this code for use with [neosr](https://github.com/muslll/neosr).
Can I confirm that using ResNet18 is as simple as changing `encoder_channels` to `[64, 128, 256, 512]`, `decoder_channels` to `128` and `backbone` to `ResNet18`, then adjusting the `self.head = nn.Sequential..... Conv(256, num_classes, kernerl_size=1))` at the end of the MAF __init__ function to `Conv(dim, num_classes, kernel_size=1))`?

I also have changed `models.resnet18(pretrained=True)` to be `models.resnet18(weights=ResNet18_Weights.DEFAULT)` and included at the beginning of the file `from torchvision.models import ResNet18_Weights`.

Using this approach, I can get training going however find the resulting validation images very noisy when compared with the same configuration and seed using U-Net or A2-FPN. I assume this is due to my implementation of the code.

I would appreciate any guidance you may be able to offer.

Kind regards,
terrainer

**Comment by DrWuHonglin** [comment]
Hello! Thank you for asking this question.
According to your description, replacing backbone with ResNet18 is exactly what you say it is.

**Comment by terrainer** [comment]
> Hello! Thank you for asking this question. According to your description, replacing backbone with ResNet18 is exactly what you say it is.

Thank you for responding so quickly. If that is correct, I will continue to train and see if the noise clears up at higher iterations.
